source/kernel/src/services/serial_mux.rs

```diff
@@ -363,17 +433,35 @@ impl IncomingMuxerTask {
             };
 
             // Great, now we have a message! Let's see if we have someone listening to this port
-            let mux = self.mux.lock().await;
-            if let Some(port) = mux.ports.as_slice().iter().find(|p| p.port == port_id) {
-                if let Some(mut wgr) = port.upstream.send_grant_exact_sync(datab.len()) {
-                    wgr.copy_from_slice(datab);
-                    wgr.commit(datab.len());
-                    debug!(port_id, len = datab.len(), "Sent bytes to port");
-                } else {
-                    warn!(port_id, len = datab.len(), "Discarded bytes, full buffer");
+            match self.pseudokeyboard {
+                Some((ref mut keymux, keyboard_port)) if port_id == keyboard_port => {
```

Comment:
hmm, yeah, i suppose the special-cased traceproto stuff is implemented as a client of sermux, without any logic in sermux itself...

another approach could be having the keyboard mux server always try to acquire the sermux keyboard port, since they're both defined in the kernel, and hardware independent. that way we don't need a separate task just to forward bytes around. WDYT?

Reply:
I feel somewhat better about your other approach! Could be configurable, but always doing it now (or taking `sermux_keyboard_port: Option<u16>` in the config or something) seems reasonable to me.